t/005-apicast-oauth.t

```diff
@@ -361,3 +362,38 @@ GET /?access_token=foobar
 --- error_code: 200
 --- response_body
 yay, upstream
+
+=== TEST 12: calling /authorize with state returns same value back on redirect_uri
```

Comment:
I guess it is fine now. It is really confusing, because we are reusing `params.state` for the nonce and that is so evil. I pushed some clarifications on the state length to indicate it is one or the other when we reuse the field.